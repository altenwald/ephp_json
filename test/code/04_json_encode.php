<?php

echo "Array sequential" . PHP_EOL;
$sequential = ["foo", "bar", "baz", "blong"];
var_dump(
    $sequential,
    json_encode($sequential)
);

echo PHP_EOL . "Array no secuential" . PHP_EOL;
$nonsequential = [1 => "foo", 2 => "bar", 3 => "baz", 4 => "blong"];
var_dump(
    $nonsequential,
    json_encode($nonsequential)
);

echo PHP_EOL . "Array secuential with destroyed key" . PHP_EOL;
unset($sequential[1]);
var_dump(
    $sequential,
    json_encode($sequential)
);
