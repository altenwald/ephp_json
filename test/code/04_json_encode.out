Array sequential
array(4) {
  [0]=>
  string(3) "foo"
  [1]=>
  string(3) "bar"
  [2]=>
  string(3) "baz"
  [3]=>
  string(5) "blong"
}
string(27) "["foo","bar","baz","blong"]"

Array no secuential
array(4) {
  [1]=>
  string(3) "foo"
  [2]=>
  string(3) "bar"
  [3]=>
  string(3) "baz"
  [4]=>
  string(5) "blong"
}
string(43) "{"1":"foo","2":"bar","3":"baz","4":"blong"}"

Array secuential with destroyed key
array(3) {
  [0]=>
  string(3) "foo"
  [2]=>
  string(3) "baz"
  [3]=>
  string(5) "blong"
}
string(33) "{"0":"foo","2":"baz","3":"blong"}"
